tests/OpenSSLErrorTest.cpp

```diff
@@ -63,9 +63,15 @@ static uint64_t fail_PEM_write_bio_RSA_PUBKEY = 0;
 static uint64_t fail_RSA_set0_key = 0;
 #endif
 
+#ifdef LIBWOLFSSL_VERSION_STRING
+#define SYMBOL_NAME(s) ("wolfSSL_" s)
```

Comment:
Do the plain openssl names not work? That seems like a breaking change 🤔

Reply:
We never exposed the same symbols as OpenSSL in our binary. I'm not sure how the tests pass without this PR. As soon as I fixed the `wolfssl/options.h` issue above this file blew up with errors and the functions would return NULL. Then the higher levels would try to call NULL which obviously segfaulted. `gdb` was very helpful to debug this:
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00005555556278a8 in wolfSSL_BIO_new (type=0x7ffff7e8a440 <meth>)
    at /home/julek/Projects/jwt-cpp/tests/OpenSSLErrorTest.cpp:80
#2  0x00007ffff7db7e0e in wolfSSL_BIO_new_mem_buf (buf=0x555555743af0, len=2118) at ./src/bio.c:2639
#3  0x00005555555ab762 in jwt::helper::make_mem_buf_bio (
    data="-----BEGIN CERTIFICATE-----\nMIIF8DCCBVmgAwIBAgIKYFOB9QABAACIvTANBgkqhkiG9w0BAQUFADBGMQswCQYD\nVQQGEwJVUzETMBEGA1UEChMKR29vZ2xlIEluYzEiMCAGA1UEAxMZR29vZ2xlIElu\ndGVybmV0IEF1dGhvcml0eTAeFw0xMzA1MjIxNTQ5MD"...)
    at /home/julek/Projects/jwt-cpp/include/jwt-cpp/jwt.h:507
#4  0x00005555555ad2ea in jwt::helper::extract_pubkey_from_cert<jwt::error::rsa_error> (
    certstr="-----BEGIN CERTIFICATE-----\nMIIF8DCCBVmgAwIBAgIKYFOB9QABAACIvTANBgkqhkiG9w0BAQUFADBGMQswCQYD\nVQQGEwJVUzETMBEGA1UEChMKR29vZ2xlIEluYzEiMCAGA1UEAxMZR29vZ2xlIElu\ndGVybmV0IEF1dGhvcml0eTAeFw0xMzA1MjIxNTQ5MD"..., pw="", 
    ec=std::error_code = {std::_V2::error_category: 0}) at /home/julek/Projects/jwt-cpp/include/jwt-cpp/jwt.h:541
#5  0x00005555555ac9ee in jwt::helper::extract_pubkey_from_cert<jwt::error::rsa_error> (
    certstr="-----BEGIN CERTIFICATE-----\nMIIF8DCCBVmgAwIBAgIKYFOB9QABAACIvTANBgkqhkiG9w0BAQUFADBGMQswCQYD\nVQQGEwJVUzETMBEGA1UEChMKR29vZ2xlIEluYzEiMCAGA1UEAxMZR29vZ2xlIElu\ndGVybmV0IEF1dGhvcml0eTAeFw0xMzA1MjIxNTQ5MD"..., pw="")
    at /home/julek/Projects/jwt-cpp/include/jwt-cpp/jwt.h:578
#6  0x00005555555a7b9a in HelperTest_Cert2Pubkey_Test::TestBody (this=0x5555557516a0)
    at /home/julek/Projects/jwt-cpp/tests/HelperTest.cpp:11
#7  0x000055555568e4ff in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) ()
#8  0x0000555555682de6 in testing::Test::Run() ()
#9  0x0000555555682f65 in testing::TestInfo::Run() ()
#10 0x0000555555683519 in testing::TestSuite::Run() ()
#11 0x0000555555683c1f in testing::internal::UnitTestImpl::RunAllTests() ()
#12 0x000055555568eac7 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) ()
#13 0x000055555568302c in testing::UnitTest::Run() ()
#14 0x00005555555b0811 in RUN_ALL_TESTS () at /usr/include/gtest/gtest.h:2490
#15 0x00005555555b078a in main (argc=1, argv=0x7fffffffdd88) at /home/julek/Projects/jwt-cpp/tests/TestMain.cpp:5
```

Reply:
Yikes. Well I am glad you knew to fox that since it was not obvious. I had hit bugs in the passed with the certificates and they were populated so I hadn't suspected anything 😅 

It might be worth adding an install of the cmake config files or maybe add these flags to the Pkg-config files?

Reply:
Which flags exactly do you mean? You can't use wolfSSL with different flags than it was compiled with since that will run into all kinds of trouble. A good solution would be to embed the defines inside of the `jwt.h` header but I'm not a cmake expert to make that happen.

Reply:
`EXTERNAL_OPTS_OPENVPN` for example.
No, the library should export the defines it was built with. Or else downstream will not know. End-users could build wolfssl with any options they like so it become problematic to hardcoded build settings here.

Reply:
That is the purpose of `wolfssl/options.h`. It contains the options used while building. `EXTERNAL_OPTS_OPENVPN` is only a convenience macro for compatibility users to not have to include `wolfssl/options.h`. Perhaps a better way would be to include `wolfssl/options.h` in `jwt.h`.

Reply:
Ahhh. I that would make more sense. The OENVPN flag doesn't carry the same meaning.

Reply:
Its poor unfortunate legacy naming I agree. I'll try to refactor this patch.

Reply:
I added `options.h` including explicitly in the last commit.